src/benchq/algorithms/_qsp.py

```diff
@@ -78,7 +78,8 @@ def get_qsp_program(
 
     if mode == "gse":
         # TODO: explain where this formula comes from
-        steps = int(np.ceil(np.pi * (pyliqtr_operator.alpha) / (gse_accuracy * 2)))
+        n_block_encodings = int(np.ceil(np.pi * (pyliqtr_operator.alpha) / (gse_accuracy * 2)))
+        steps = n_block_encodings*2 + 3 # *2 for each layer consisting of 2 blocks, +2 for rotation layers, +1 for extra select-V
     elif mode == "time_evolution":
```

Comment:
I'm a little confused here. Are the steps all block encodings? Why are the steps performing different tasks? (rotation and select-v)